source/extensions/quic_listeners/quiche/udp_gso_batch_writer.cc

```diff
@@ -0,0 +1,129 @@
+    : quic::QuicGsoBatchWriter(std::make_unique<quic::QuicBatchWriterBuffer>(), io_handle.fd()),
+      stats_(generateStats(scope)) {}
+
+// Do Nothing in the Destructor For now
+UdpGsoBatchWriter::~UdpGsoBatchWriter() = default;
+
+Api::IoCallUint64Result
+UdpGsoBatchWriter::writePacket(const Buffer::Instance& buffer, const Network::Address::Ip* local_ip,
+                               const Network::Address::Instance& peer_address) {
+  // Convert received parameters to relevant forms
+  quic::QuicSocketAddress peer_addr = envoyAddressIpToQuicSocketAddress(peer_address.ip());
+  quic::QuicSocketAddress self_addr = envoyAddressIpToQuicSocketAddress(local_ip);
+
+  // TODO(yugant): Currently we do not use PerPacketOptions with Quic, we may want to
+  // specify this parameter here at a later stage.
+  quic::WriteResult quic_result = WritePacket(
+      buffer.toString().c_str(), static_cast<size_t>(buffer.length()), self_addr.host(), peer_addr,
+      /*quic::PerPacketOptions=*/nullptr);
+  updateUdpPacketWriterStats(quic_result);
+
+  return convertQuicWriteResult(quic_result);
+}
+
+uint64_t UdpGsoBatchWriter::getMaxPacketSize(const Network::Address::Instance& peer_address) const {
+  quic::QuicSocketAddress peer_addr = envoyAddressIpToQuicSocketAddress(peer_address.ip());
+  return static_cast<uint64_t>(GetMaxPacketSize(peer_addr));
+}
+
+Network::UdpPacketWriterBuffer
+UdpGsoBatchWriter::getNextWriteLocation(const Network::Address::Ip* local_ip,
+                                        const Network::Address::Instance& peer_address) {
+  quic::QuicSocketAddress peer_addr = envoyAddressIpToQuicSocketAddress(peer_address.ip());
+  quic::QuicSocketAddress self_addr = envoyAddressIpToQuicSocketAddress(local_ip);
+  quic::QuicPacketBuffer quic_buf = GetNextWriteLocation(self_addr.host(), peer_addr);
+  return Network::UdpPacketWriterBuffer(quic_buf.buffer, quic_buf.release_buffer);
+}
+
+Api::IoCallUint64Result UdpGsoBatchWriter::flush() {
+  quic::WriteResult quic_result = Flush();
+  updateUdpPacketWriterStats(quic_result);
+
+  return convertQuicWriteResult(quic_result);
+}
+
+void UdpGsoBatchWriter::updateUdpPacketWriterStats(quic::WriteResult quic_result) {
+  if (quic_result.status == quic::WRITE_STATUS_OK && quic_result.bytes_written > 0) {
+    if (stats_.front_buffered_pkt_size_.value() > 0u) {
+      // NOTE: int division with uint64_t. This is safe as the int term
+      // will always be greater than zero here.
+      uint64_t num_pkts_in_batch =
+          quic_result.bytes_written / stats_.front_buffered_pkt_size_.value();
+      // If current pkt was also sent, then internal buffer would be empty
+      bool current_pkt_sent_with_batch = (batch_buffer().SizeInUse() == 0u);
+      stats_.pkts_sent_per_batch_.recordValue(current_pkt_sent_with_batch ? num_pkts_in_batch + 1u
+                                                                          : num_pkts_in_batch);
```

Comment:
Yes, it does the same! Initially, I was using the same method to calculate the number of packets per batch. But decided against it because it involved typecasting and using `std::ceil`. 

Calculating `num_pkts_in_batch` using `std::ceil` now!